packages/baseui-codemods/src/v9-theme-scale.test.js

```diff
@@ -0,0 +1,256 @@
+  describe('does update', () => {
+    it('font values in Blocks', async () => {
+      const content = `const Foo = () => (
+    <React.Fragment>
+      <Block font="font1100" />
+      <Block font="font1000" />
+      <Block font="font900" />
+      <Block font="font800" />
+      <Block font="font700" />
+      <Block font="font600" />
+      <Block font="font500" />
+      <Block font="font470" />
+      <Block font="font460" />
+      <Block font="font450" />
+      <Block font="font400" />
+      <Block font="font350" />
+      <Block font="font300" />
+      <Block font="font250" />
+      <Block font="font200" />
+      <Block font="font100" />
+    </React.Fragment>
+  );`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = () => (
+            <React.Fragment>
+              <Block font=\\"font1450\\" />
+              <Block font=\\"font1050\\" />
+              <Block font=\\"font950\\" />
+              <Block font=\\"font850\\" />
+              <Block font=\\"font750\\" />
+              <Block font=\\"font650\\" />
+              <Block font=\\"font550\\" />
+              <Block font=\\"font450\\" />
+              <Block font=\\"font400\\" />
+              <Block font=\\"font350\\" />
+              <Block font=\\"font300\\" />
+              <Block font=\\"font250\\" />
+              <Block font=\\"font200\\" />
+              <Block font=\\"font150\\" />
+              <Block font=\\"font100\\" />
+              <Block font=\\"font100\\" />
+            </React.Fragment>
+          );"
+      `);
+    });
+
+    it('direct usage of font values', async () => {
+      const content = `const Foo = styled('div', props => ({
+    ...props.$theme.typography.font300
+  }));`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = styled('div', props => ({
+            ...props.$theme.typography.font200
+          }));"
+      `);
+    });
+
+    it('direct usage of values with destructured `$theme`', async () => {
+      const content = `const Foo = styled('div', ({$theme}) => ({
+    ...$theme.typography.font300
+  }));`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = styled('div', ({$theme}) => ({
+            ...$theme.typography.font200
+          }));"
+      `);
+    });
+
+    it('direct usage of values with destructured value', async () => {
+      const content = `const Foo = styled('div', ({$theme: {typography: {font300}}}) => ({
+    ...$theme.typography.font300
+  }));`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = styled('div', ({$theme: {typography: {font200}}}) => ({
+            ...$theme.typography.font200
+          }));"
+      `);
+    });
+
+    it('override style function', async () => {
+      const content = `const Foo = <Block overrides={{ Block: { style: ({$theme}) => ({ ...$theme.typography.font200 })}}} />`;
```

Comment:
does this work for other baseui components besides block?

Reply:
yeah this should be picked up by the identifier traversal - but add change it for another component